lisa/tests/staging/scheduler/util_tracking.py

```diff
@@ -0,0 +1,338 @@
+    `util_avg` value at the end of each activation.
+
+    Two signals composes the estimated utlization of a task:
+
+    * `util_est_enqueued` : is expected to match the max between `util_avg` and
+      `util_est_ewma` at the end of the previous activation
+
+    * `util_est_ewma` : is expected to track an Exponential Weighted Moving
+      Average of the `util_avg` signal sampled at the end of each activation.
+
+    Based on these two invariant, this class provides a set of tests to verify
+    these conditions using different methods and sampling points.
+    """
+
+    PhaseStats = namedtuple("PhaseStats", [
+        'start', 'end', 'area_util', 'area_enqueued', 'area_ewma'])
+
+    ActivationSignals = namedtuple("ActivationSignals", [
+        'time', 'util_avg', 'util_est_enqueued', 'util_est_ewma'])
+
+    @classmethod
+    def get_rtapp_profile(cls, plat_info):
+        big_cpu = plat_info["capacity-classes"][-1][0]
+
+        task = (
+            # Big task: 15 activations at full utilization
+            Periodic(
+                duty_cycle_pct=75,
+                duration_s=5,
+                period_ms=200,
+                cpus=[big_cpu]) +
+            # Ramp Down
+            Ramp(
+                start_pct=50,
+                end_pct=5,
+                delta_pct=20,
+                time_s=1,
+                period_ms=200,
+                cpus=[big_cpu]) +
+            # Ramp Up
+            Ramp(
+                start_pct=10,
+                end_pct=60,
+                delta_pct=20,
+                time_s=1,
+                period_ms=200,
+                cpus=[big_cpu])
+        )
+        return {'test_task' : task}
+
+    @property
+    @memoized
+    def fast_ramp(self):
+        # The EWMA "fast ramp up" feature has been merged only in kernel TBD
+        # TODO fix kernel version
+        return False
+
+        min_kernel = KernelVersion('4.5-rc1').parts
+        cur_kernel = self.plat_info['kernel']['version'].parts
+        return cur_kernel < min_kernel
+
+    def _plot_signals(self, task, test, failures):
+        signals = ['util', 'util_est_enqueued', 'util_est_ewma']
+        ax = self.trace.analysis.load_tracking.plot_task_signals(task, signals=signals)
+        ax = self.trace.analysis.rta.plot_phases(task, axis=ax);
+        for start in failures:
+            ax.axvline(start, alpha=0.5, color='r', axis=ax)
+        filepath = os.path.join(self.res_dir, 'util_est_{}.png'.format(test))
+        self.trace.analysis.rta.save_plot(ax.figure, filepath=filepath)
+
+    @requires_events('sched_util_est_task', 'sched_load_se')
+    @RTAEventsAnalysis.task_phase_windows.used_events
+    @RTATestBundle.check_noisy_tasks(noise_threshold_pct=1)
+    def test_areas(self) -> ResultBundle:
+        """
+        Test signals are properly "dominated".
+
+        The integral of `util_est_enqueued` is expected to be always not
+        smaller than that of `util_avg`, since this last is subject to decays
```

Comment:
I have only one negation unit in my sentence reading pipeline, what about `expected to be always greater than that of...`